src/time_machine/__init__.py

```diff
@@ -121,6 +127,12 @@ def extract_timestamp_tzname(
     elif isinstance(dest, dt.datetime):
         if HAVE_ZONEINFO and isinstance(dest.tzinfo, ZoneInfo):
             tzname = dest.tzinfo.key
+        elif HAVE_PYTZ and isinstance(dest.tzinfo, pytz.BaseTzInfo):
+            raise RuntimeError(
```

Comment:
Use `TypeError` because it's a bad *type* of `tzinfo`

```suggestion
            raise TypeError(
```